### ISSUE TITLE
AIP-38 Programmatic form component

### ISSUE THREAD

**Issue by pierrejeambrun** [apache/airflow#42372]
Brent says:
> Let's develop a robust form component that can render many different fields based on the spec it receives. Then we > can use this for advanced dagrun trigger and connection management.

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

**Comment by jscheffl** [comment]
I would try to raise my hand for this. Just need a bit time to get started in the new ecosystem :-D 

THe components woul dbe needed for (1) trigger form as well as (2) connection for in the feature set of the legacy UI.

**Comment by pierrejeambrun** [comment]
Yes the idea is to have a generic form component, flexible, reusable enough to do any type of form. A good idea would be to take a look into react form libraries out there and find one that looks good. (input validation, error handling, etcc...). Back in the time I used formik, I don't know if it is still relevent today or if other libraries are better suited.

**Comment by bbovenzi** [comment]
Yes, I know formik is very popular: https://formik.org/ and Chakra already has docs on how to integrate with it: https://v2.chakra-ui.com/getting-started/with-formik

**Comment by bbovenzi** [comment]
TanStack also has a new form library: https://tanstack.com/form/latest
We already use their libraries for queries and tables. so definitely worth a look as well.

**Comment by bbovenzi** [comment]
Updating my comment based on some quick research.

formik: https://formik.org/ Has had long periods of not being maintained, not typescript-first
tanstack/form: https://tanstack.com/form/latest His other libraries, query and table, have been very useful for us, but its still at v0.33.0 so it could change a lot after a v1.0
react-hook-form: https://react-hook-form.com/ Another popular framework.

I would lean towards react-hook-form, idk if we want to rely on an outdated library or anything at a 0.X version

**Comment by pierrejeambrun** [comment]
I share the feeling for `0.X` versions. This is something I mentioned when we considered switching the backend web framework, but still we went with FastAPI, so I would say for consistancy that 0.X version are not a problem for us. (I guess)
